Spigot-Server-Patches/0581-Improve-ServerGUI.patch

```diff
@@ -0,0 +1,453 @@
+ 
++        // Paper start - Add tps entry
++        double[] tps = org.bukkit.Bukkit.getTPS();
++        String[] tpsAvg = new String[tps.length];
++
++        for ( int g = 0; g < tps.length; g++) {
++            tpsAvg[g] = format( tps[g] );
++        }
+         this.d[0] = "Memory use: " + i / 1024L / 1024L + " mb (" + Runtime.getRuntime().freeMemory() * 100L / Runtime.getRuntime().maxMemory() + "% free)";
+         this.d[1] = "Avg tick: " + GuiStatsComponent.a.format(this.a(this.e.h) * 1.0E-6D) + " ms";
++        getStatEntries()[2] = "TPS from last 1m, 5m, 15m: " + String.join(", ", tpsAvg);
++        // Paper end
++
+         this.b[this.c++ & 255] = (int) (i * 100L / Runtime.getRuntime().maxMemory());
+         this.repaint();
+     }
+@@ -83,4 +93,10 @@ public class GuiStatsComponent extends JComponent {
+     public void a() {
+         this.f.stop();
+     }
++
++    public String[] getStatEntries() { return this.d; } // Paper - OBFHELPER
++    private static String format(double tps) // Paper - Add tps entry
++    {
++        return ( ( tps > 21.0 ) ? "*" : "" ) + Math.min( Math.round( tps * 100.0 ) / 100.0, 20.0 ); // Paper - only print * at 21, we commonly peak to 20.02 as the tick sleep is not accurate enough, stop the noise
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/ServerGUI.java b/src/main/java/net/minecraft/server/ServerGUI.java
+index e51434f2323bcb9385d11e99d0438d66434e3e63..032283c520536cafa075dfd3690590c03e488b94 100644
+--- a/src/main/java/net/minecraft/server/ServerGUI.java
++++ b/src/main/java/net/minecraft/server/ServerGUI.java
+@@ -2,15 +2,15 @@ package net.minecraft.server;
+ 
+ import com.google.common.collect.Lists;
+ import com.mojang.util.QueueLogAppender;
+-import java.awt.BorderLayout;
+-import java.awt.Component;
+-import java.awt.Dimension;
+-import java.awt.Font;
++
++import java.awt.*;
+ import java.awt.event.FocusAdapter;
+ import java.awt.event.FocusEvent;
+ import java.awt.event.WindowAdapter;
+ import java.awt.event.WindowEvent;
++import java.io.IOException;
+ import java.util.Collection;
++import java.util.Objects;
+ import java.util.concurrent.atomic.AtomicBoolean;
+ import javax.swing.JComponent;
+ import javax.swing.JFrame;
+@@ -22,6 +22,7 @@ import javax.swing.JTextArea;
+ import javax.swing.JTextField;
+ import javax.swing.SwingUtilities;
+ import javax.swing.UIManager;
++import javax.imageio.ImageIO;
+ import javax.swing.border.EtchedBorder;
+ import javax.swing.border.TitledBorder;
+ import javax.swing.text.AttributeSet;
+@@ -29,6 +30,7 @@ import javax.swing.text.BadLocationException;
+ import javax.swing.text.Document;
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
++import sun.awt.X11.XToolkit;
+ 
+ public class ServerGUI extends JComponent {
+ 
+@@ -46,6 +48,16 @@ public class ServerGUI extends JComponent {
+             ;
+         }
+ 
++        // Paper start - Fix windows title issue on gnome desktop
```

Comment:
You don't need reflection nor sun pkg stuff here. Did you not look at the stackoverflow answer I linked?